### PR TITLE
Use new components endpoint in frontend

### PR DESCRIPTION
**Pull request by maximi4321**
# Summary

Use the new endpoints from the backend and  fix some compatibility issues with the new provided data.

## Type of change

- Refactoring.

## Changes

- add `api/component.ts` to make requests to the "component" endpoint.
- The "component" endpoint is used to retrieve tasks in `datasets/SelectTaskStep.jsx` and `experiments/SetNameAndTaskStep.jsx`
- The "component" endpoint is used to retrieve dataloaders in `datasets/SelectDataloaderStep.jsx`.
- The "component" endpoint is used to retrieve models compatible with the selected task in `experiments/ConfigureModelsStep.jsx`
- The "component" endpoint is used to retrieve models that are children of another object in `ConfigurableObject/Inputs/ClassInput.jsx` and `utils/values.jsx`
- Compatibility issues with the new data are fixed in all components that utilize the "component" endpoint.
## How to Test

1. Go to `DashAI/DashAI/front` and run `yarn build`
2. Go back to `DashAI/` and run `python -c "import DashAI;DashAI.run()"`
3. Follow the steps in the video in the "Screenshots" section

## Screenshots

[endpoints.webm](https://github.com/DashAISoftware/DashAI/assets/53657198/55a93402-a473-4c2f-8b5e-1d91fef4b941)

## Notes

There is a small modification in the JSON schema of `numericalWrapperForText`. This is only done to test the frontend and will be reverted before merging."

### CODE REVIEW

File: DashAI/front/src/api/component.ts [hunk 1]
```diff
@@ -0,0 +1,52 @@
+import api from "./api";
+import type { IDataloader } from "../types/dataloader";
+import type { ITask } from "../types/task";
+
+interface componentInput {
```
Comment (pbadillatorrealba) [inline DashAI/front/src/api/component.ts]: Do you agree if we call this interface `componentQuery`?

```suggestion
interface componentQuery{
```

File: DashAI/front/src/api/component.ts [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    const componentParentQuery = `component_parent=${componentParent}`;
+    queryArray.push(componentParentQuery);
+  }
+
+  // join all the queries strings into a single one
+  const fullQuery = queryArray.join("&");
+
+  const response = await api.get<Array<ITask | IDataloader | object>>(
+    `/v1/component/?${fullQuery}`,
+  );
```
Comment (pbadillatorrealba) [inline DashAI/front/src/api/component.ts]: Why not to use directly the axios query params in get?

File: DashAI/front/src/components/experiments/ConfigureModelsStep.jsx [hunk 3]
```diff
@@ -14,25 +15,53 @@ import uuid from "react-uuid";
...
  */
+
+const compatibleModelsTest = [
+  { name: "KNeighborsClassifier" },
+  { name: "RandomForestClassifier" },
+  { name: "SVC" },
+  { name: "numericalWrapperForText" },
+  { name: "tcTransformerEngSpa" },
+];
+
```
Comment (pbadillatorrealba) [inline DashAI/front/src/components/experiments/ConfigureModelsStep.jsx]: This should be eliminated?

File: DashAI/front/src/components/experiments/ConfigureModelsStep.jsx [hunk 4]
```diff
@@ -14,25 +15,53 @@ import uuid from "react-uuid";
...
+  //     if (error.response) {
+  //       console.error("Response error:", error.message);
+  //     } else if (error.request) {
+  //       console.error("Request error", error.request);
+  //     } else {
+  //       console.error("Unkown Error", error.message);
+  //     }
+  //   }
+  // };
+
```
Comment (pbadillatorrealba) [inline DashAI/front/src/components/experiments/ConfigureModelsStep.jsx]: I think this should be de-commented.

**Review by pbadillatorrealba — CHANGES_REQUESTED** [review]
Some initial comments about the API and commented code.

File: DashAI/front/src/api/component.ts (hunk 2 above)
Comment (maximi4321) [reply DashAI/front/src/api/component.ts]: i updated the function to use axios query params

File: DashAI/front/src/components/experiments/ConfigureModelsStep.jsx (hunk 4 above)
Comment (maximi4321) [reply DashAI/front/src/components/experiments/ConfigureModelsStep.jsx]: Yes, i was using a test version of the data to be able to test all the models.
In the latest commit i am now using the endpoint

**Review by cristian-tamblay — APPROVED** [review]

**Review by pbadillatorrealba — APPROVED** [review]

**Pull request merged**